src/cluster.rs

```diff
@@ -287,3 +291,102 @@ impl TryFrom<crate::xds::config::endpoint::v3::ClusterLoadAssignment> for Cluste
         })
     }
 }
+
+/// Set around [`LocalityEndpoints`] to ensure that all unique localities are
+/// different entries. Any duplicate localities provided are merged.
+#[derive(Clone, Default, Debug, Eq, PartialEq)]
+pub struct LocalitySet(HashMap<Option<Locality>, LocalityEndpoints>);
```

Comment:
When will the key not be `Some<Locality>` and instead be `None`? Just to be meet requirements of the deserialization?

Reply:
The xDS management server can return endpoints with no locality, also for configuration it makes more sense to allow `null` for locality, as not everyone is going to care about locality. It's also not used presently, it's just there because it's easier to align with xDS.

https://github.com/googleforgames/quilkin/blob/ac7dc87c3f7828c496dff2d416a443411f879c48/src/endpoint/locality.rs#L133-L145